File: anyhedge/utils/auth_token.py
from ..js.runner import AnyhedgeFunctions

def get_settlement_service_auth_token(scheme, domain, port, raise_exception=False):
    try:
        return AnyhedgeFunctions.getSettlementServiceAuthToken({
            "scheme": scheme,
            "domain": domain,
            "port": port,
        })
    except Exception as exception:
        if raise_exception:
            raise exception

File: anyhedge/js/src/funcs/authToken.js
import { AnyHedgeManager } from "@generalprotocols/anyhedge";

/**
 *
 * @param {Object} settlementService 
 * @param {String} settlementService.scheme 
 * @param {String} settlementService.domain 
 * @param {String} settlementService.port 
 */
export async function getSettlementServiceAuthToken(settlementService) {
  const manager = new AnyHedgeManager({
    serviceScheme: settlementService.scheme,
    serviceDomain: settlementService.domain,
    servicePort: settlementService.port,
  })
  return await manager.requestAuthenticationToken('Paytaca')
}
